src/app/pages/send/ordinal-inscription/components/create-utxo-from-inscription.ts

```diff
@@ -1,26 +1,25 @@
-import type { UtxoWithDerivationPath } from '@leather-wallet/query';
+import type { InscriptionResponse, UtxoWithDerivationPath } from '@leather-wallet/query';
 
 import { BitcoinNetworkModes } from '@shared/constants';
 import { getNativeSegwitAddressIndexDerivationPath } from '@shared/crypto/bitcoin/p2wpkh-address-gen';
-import { Inscription } from '@shared/models/inscription.model';
 
 interface CreateUtxoFromInscriptionArgs {
-  inscription: Inscription;
+  inscriptionResponse: InscriptionResponse;
   network: BitcoinNetworkModes;
   accountIndex: number;
 }
 
 export function createUtxoFromInscription({
-  inscription,
+  inscriptionResponse,
   network,
   accountIndex,
 }: CreateUtxoFromInscriptionArgs): UtxoWithDerivationPath {
   const { genesis_block_hash, genesis_timestamp, genesis_block_height, value, addressIndex } =
```

Comment:
I see problem here. `addressIndex` was added/decorated to the type in the extension, as it's needed later on, but it isn't in response as you can only know that if you're generating the address.

Should be removed from here

Reply:
Yeah, an inscription is basically a decorated UTXO with extra metadata.

See this code here where we introduce the `addressIndex` to the `Inscription` type 

https://github.com/leather-wallet/extension/blob/effefd5f25f95fa723ceb5f5d4a725df74c85113/src/app/query/bitcoin/ordinals/inscriptions.query.ts#L179-L182

We do this because, if we want to spend/send it later to someone else, we need to know which address to sign with.

Now that we're re-doing the models though, I think they should be separate. An inscription coming from the API doesn't know it's owner, so the core type shouldn't include `addressIndex` (nor is it being an optional param a good idea).

I'd create a new type, that contains `Inscription` if we need to add additional wallet-only metadaata, e.g. 

```ts
interface InscriptionWithOwner {
  inscription: Inscription;
  ownerPath: string;
}
const inscriptionWithOwner = { inscription, path: 'm/86'/0'/3'/0/0' } satisfies InscriptionWithOwner;
```

Not sold on naming here, but composing a new type with ownership is better than adding conditionality to core type.